### PR TITLE
Update the CI and README for Python 3.7

### PR DESCRIPTION
**Pull request by jeremycline**
This updates the Tox configuration, Travis CI configuration, and notes Python 3.7 is supported in the README and classifiers.

Fixes: #117

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,27 +1,45 @@
 language: python
-
-env:
```
Comment (itamarst) [inline .travis.yml]: I think you can keep the old, shorter, version by adding `dist: xenial` at the top level, as in https://github.com/itamarst/eliot/blob/master/.travis.yml#L2

**Comment by itamarst** [comment]
Oh, and of course—thank you for doing this, really appreciate it!

File: .travis.yml (hunk 1 above)
Comment (jeremycline) [reply .travis.yml]: Okay, I'll give that a shot. I *think* it's going to mash the two Twisted versions together in a single sub-test unless I pull the Twisted selection stuff back out of the Tox setup (which would make it a little harder to run the same tests locally), but I'll see.

Comment (itamarst) [reply .travis.yml]: Ah, then maybe also do what I did in Eliot and use `tox-travis`?

Make sure tox-travis is installed:
https://github.com/itamarst/eliot/blob/master/.travis.yml#L9

Add a mapping from Travis Python version to the tox commands to run:
https://github.com/itamarst/eliot/blob/master/tox.ini#L6

**Comment by jeremycline** [comment]
Okay, docs were read, assumptions were broken, and I believe the last build should correctly run one Tox environment per Travis job and be less hideous than version 1 and local tox runs also appear to work correctly. I also added a news snippet. Thanks for the review and sorry about all the build noise!

**Comment by itamarst** [comment]
Looks good, thank you!

**Comment by itamarst** [comment]
OK, I've released 1.10.

**Pull request merged**